src/zenml/integrations/gcp/secrets_manager/gcp_secrets_manager.py

```diff
@@ -176,102 +371,135 @@ def get_secret(self, secret_name: str) -> BaseSecretSchema:
                 secret_value = response.payload.data.decode("UTF-8")
 
                 secret_key = remove_group_name_from_key(
-                    secret.name.split("/")[-1], secret_name
+                    google_secret.name.split("/")[-1], secret_name
                 )
 
                 secret_contents[secret_key] = secret_value
 
-                zenml_schema_name = secret.labels[ZENML_SCHEMA_NAME]
+                zenml_schema_name = google_secret.labels[ZENML_SCHEMA_NAME]
 
-        if not secret_contents:
-            raise RuntimeError(f"No secrets found within the {secret_name}")
+            if not secret_contents:
+                raise KeyError(
+                    f"Can't find the specified secret '{secret_name}'"
+                )
 
-        secret_contents["name"] = secret_name
+            secret_contents["name"] = secret_name
 
-        secret_schema = SecretSchemaClassRegistry.get_class(
-            secret_schema=zenml_schema_name
-        )
-        return secret_schema(**secret_contents)
+            secret_schema = SecretSchemaClassRegistry.get_class(
+                secret_schema=zenml_schema_name
+            )
+            zenml_secret = secret_schema(**secret_contents)
+
+        else:
+            # Scoped secrets are mapped 1-to-1 with Google secrets
+
+            google_secret_name = self.CLIENT.secret_path(
+                self.project_id, self._get_scoped_secret_name(secret_name)
+            )
+
+            try:
+                # fetch the latest secret version
+                google_secret = self.CLIENT.get_secret(name=google_secret_name)
+            except google_exceptions.NotFound:
+                raise KeyError(
+                    f"Can't find the specified secret '{secret_name}'"
+                )
+
+            # make sure the secret has the correct scope labels to filter out
+            # unscoped secrets with similar names
+            scope_labels = self._get_secret_scope_metadata(secret_name)
+            if not scope_labels.items() <= google_secret.labels.items():
```

Comment:
I have never seen that syntax of comparing dict items, can you explain what it does here?

Reply:
This is some python 3 magic. It checks that all the items in the scope_labels dictionary are included in the google secrets labels. I added a comment to clarify.

The `dict_items` type behaves like a set: all of the operations defined for the abstract base class collections.abc.Set are available (for example, ==, <, or ^). More here: https://docs.python.org/3/library/stdtypes.html#dict-views

Reply:
I see, a set of tuples and then it's a subset check, interesting. Thanks!